### ISSUE TITLE
"Unexpected error! - error Error Code" when trying to upload shapefile

### ISSUE THREAD

**Issue by mcarson75** [GeoNode/geonode#7328]
## Expected Behavior
Shapefile upload successful

## Actual Behavior
Receive this error code on shapefile upload
`Unexpected error! - error Error Code. Contact the system administrator for more information regarding this error message.`

## Steps to Reproduce the Problem

  1. Upload ESRI Shapefile

## Specifications

  - GeoNode version: 3.2.0
  - Installation method (manual, GeoNode Docker, SPCGeoNode Docker): Geonode Docker
  - Platform: Ubuntu 20.04 on Azure
  - Additional details: I followed the installation instructions at https://docs.geonode.org/en/master/install/advanced/core/index.html#docker.  We are unable to upload a Shapefile (other file types work correctly.)  When I connect to django4geonode and check /var/log/geonode.log, I see many of the following errors:

```
Invalid HTTP_HOST header: '127.0.0.1:8001'. You may need to add '127.0.0.1' to ALLOWED_HOSTS.
[pid: 346|app: 0|req: 12/29] 127.0.0.1 () {24 vars in 254 bytes} [Sat Apr 17 22:09:54 2021] GET / => generated 26 bytes in 0 msecs (HTTP/1.1 400) 2 headers in 67 bytes (1 switches on core 0)
```

I modified ALLOWED_HOSTS in the .env file from the default of just django to include localhost and geonode following an examination of settings.py, but this did not change the behavior.

```
#ALLOWED_HOSTS=['django', '*']
ALLOWED_HOSTS=['django', 'localhost', 'geonode', '*']
```

**Comment by t-book** [comment]
@mcarson75 please set for testing debug to true. down and up the stack again and double check the geoserver credentials in .env

**Comment by mcarson75** [comment]
@t-book Thank you for your response!  I did resolve the `Invalid HTTP_HOST` error by adding 127.0.01 to the ALLOWED_HOSTS variable in the django settings of my site-specific docker-compose.yml file.  However, this may have been a red herring because, although I no longer see that error in the logs, I still get the `Unexpected error!` when I try to upload a shapefile.

I did verify that `GEOSERVER_ADMIN_PASSWORD` in .env is correct, and I set `DEBUG=TRUE` and rebuilt the containers.

I looked in `/var/log/geonode.log` in the django container and did not see any errors, or anything more descriptive with DEBUG enabled.  Is there a different log I should be looking in?
